go-app-ussd_nurse.js

```diff
@@ -72,6 +82,82 @@ go.app = function() {
             });
         };
 
+        self.jembi_nc_clinic_validate = function (im, clinic_code) {
+            var params = {
+                'criteria': 'value:' + clinic_code
+            };
+            return self
+            .jembi_json_api_call('get', params, null, 'NCfacilityCheck', im);
+        };
+
+        self.validate_nc_clinic_code = function(im, clinic_code) {
+            if (!utils.check_valid_number(clinic_code) ||
+                clinic_code.length !== 6) {
+                return Q()
+                    .then(function() {
+                        return false;
+                    });
+            } else {
+                return self
+                .jembi_nc_clinic_validate(im, clinic_code)
+                .then(function(json_result) {
+                    var rows = json_result.data.rows;
+                    if (rows.length === 0) {
+                        return false;
+                    } else {
+                        return rows[0][2];
+                    }
+                });
+            }
+        };
+
+        self.jembi_json_api_call = function(method, params, payload, endpoint, im) {
+            var http = new JsonApi(im, {
+                auth: {
+                    username: im.config.jembi.username,
+                    password: im.config.jembi.password
+                }
+            });
+            switch(method) {
+                case "post":
```

Comment:
I think we can drop this now because we don't post to Jembi from Jsbox apps anymore